File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/Action.java
package com.thewizrd.shared_resources.helpers;

public abstract class Action {
    protected Actions action;
    boolean actionSuccessful;
    private ActionStatus status;

    Action() {
        actionSuccessful = true;
        status = ActionStatus.UNKNOWN;
    }

    public boolean isActionSuccessful() {
        return actionSuccessful;
    }

    public void setActionSuccessful(boolean success) {
        actionSuccessful = success;
    }

    public void setActionSuccessful(ActionStatus status) {
        actionSuccessful = (status == ActionStatus.SUCCESS);
        this.status = status;
    }

    public ActionStatus getActionStatus() {
        return status;
    }

    public Actions getAction() {
        return action;
    }

    public Action(Actions action) {
        this();
        this.action = action;
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/Action.kt
package com.thewizrd.shared_resources.helpers

abstract class Action {
    var isActionSuccessful = true

    fun setActionSuccessful(status: ActionStatus) {
        isActionSuccessful = status == ActionStatus.SUCCESS
        actionStatus = status
    }

    var actionStatus: ActionStatus
        private set

    var actionType: Actions
        protected set

    constructor(_action: Actions) {
        this.actionType = _action
        isActionSuccessful = true
        actionStatus = ActionStatus.UNKNOWN
        this.actionType = _action
    }
}